### PR TITLE
refactor(balancer): get headers as hash identifier by ngx.var

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Before this PR we use `ngx.req.get_headers()` to get all possible header values as identifier,
but since nginx 1.23.0 or OpenResty  1.25.3.1 `ngx.var` can get a combined value for all header values 
with identical name (joined by comma), so I think that we could simplify these code. 

KAG-4572

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

File: kong/runloop/balancer/init.lua [hunk 1]
```diff
@@ -130,10 +130,9 @@ local function get_value_to_hash(upstream, ctx)
       identifier = var.remote_addr
 
     elseif hash_on == "header" then
-      identifier = ngx.req.get_headers()[upstream[header_field_name]]
-      if type(identifier) == "table" then
-        identifier = table_concat(identifier)
-      end
+      -- since nginx 1.23.0/openresty 1.25.3.1
+      -- ngx.var will automatically combine all header values with identical name
```
Comment (oowl) [inline kong/runloop/balancer/init.lua:134]: Can you give us a URL to verify it?

Comment (chronolaw) [reply kong/runloop/balancer/init.lua:134]: This PR has added a test case for it.

Comment (oowl) [reply kong/runloop/balancer/init.lua:134]: OK, gotcha!

**Review by oowl — APPROVED** [review]

**Comment by bungle** [comment]
@chronolaw please use this instead and pass the `ctx` that is already there:
https://github.com/Kong/kong/blob/master/kong/tools/http.lua#L533

**Review by bungle — CHANGES_REQUESTED** [review]
see above

**Review by bungle — APPROVED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**